controllers/service/eventhandlers/service_events.go

```diff
@@ -58,6 +60,9 @@ func (h *enqueueRequestsForServiceEvent) Generic(e event.GenericEvent, queue wor
 }
 
 func (h *enqueueRequestsForServiceEvent) isServiceSupported(service *corev1.Service) bool {
+	if h.handleByDefault {
```

Comment:
this will apply for all service type without considering the load balancer type annotation values.